### PR TITLE
Fix randomness fetching around null blocks

### PR DESCRIPTION
**Pull request by arajasek**
This is the larger but "better" fix to the critical portion of #3613. 

- [x] Needs tests
- [ ] Can we come up with a creative way to avoid the height check in the Chain API impl? That's the only bit I'm unhappy about.
- [ ] Confirm we should actually be changing ChainRandomness fetching too.

### CODE REVIEW

File: node/impl/full/chain.go [hunk 1]
```diff
@@ -95,7 +97,12 @@ func (a *ChainAPI) ChainGetRandomnessFromTickets(ctx context.Context, tsk types.
 		return nil, xerrors.Errorf("loading tipset key: %w", err)
 	}
 
-	return a.Chain.GetChainRandomness(ctx, pts.Cids(), personalization, randEpoch, entropy)
+	// Doing this here is slightly nicer than doing it in the chainstore directly, but it's still bad for ChainAPI to reason about network upgrades
+	if randEpoch > build.UpgradeHyperdriveHeight {
```
Comment (arajasek) [inline node/impl/full/chain.go:101]: Would love a creative suggestion that allows us to not do this

**Comment by Kubuxu** [comment]
This version seems better to me. The Chain API having to check network version is annoying, especially if we consider that in future upgrades might depend on state and not a pure chain.

**Comment by arajasek** [comment]
Added a test which will pass once we make a minor tweak to the tipset skipcache index (waiting on some input from Why)

File: chain/sync_test.go [hunk 2]
```diff
@@ -833,3 +891,55 @@ func TestSyncCheckpointEarlierThanHead(t *testing.T) {
 	require.Equal(tu.t, p1Head, a.TipSet())
 	tu.assertBad(p1, b.TipSet())
 }
+
+func TestDrandNull(t *testing.T) {
```
Comment (magik6k) [inline chain/sync_test.go:895]: Seems to be failing on CI - https://app.circleci.com/pipelines/github/filecoin-project/lotus/14479/workflows/adc1f7a9-40b6-41f2-a72c-b9cb00f20c70/jobs/151579

Comment (arajasek) [reply chain/sync_test.go:895]: Yes, because of the tipset cache thing i mentioned above^

**Review by Kubuxu — APPROVED** [review]
SGWM, regarding the ugliness: the randomness just does not belong in Chain API,
as choosing randomness turns out to be a consensus rule built on top of the content of the chain.

File: chain/sync_test.go [hunk 3]
```diff
@@ -833,3 +891,55 @@ func TestSyncCheckpointEarlierThanHead(t *testing.T) {
...
+	H := 10
+	v5h := abi.ChainEpoch(50)
+	build.UpgradeHyperdriveHeight = v5h
```
Comment (arajasek) [inline chain/sync_test.go:898]: mental note to reset this at end of test

**Review by magik6k — APPROVED** [review]

**Pull request merged**